src/main/app/src/app/admin/smart-documents.service.ts

```diff
@@ -9,6 +9,18 @@ import { FoutAfhandelingService } from "../fout-afhandeling/fout-afhandeling.ser
 import { ZacHttpClient } from "../shared/http/zac-http-client";
 import { GeneratedType } from "../shared/utils/generated-types";
 
+export type SmartDocumentsTemplateWithParentId =
```

Comment:
What do we need this type for? Is it not generated and should therefore not be committed to the code ?

Reply:
It is used for creating type `SmartDocumentsTemplateGroupWithParentId`. But I can also do that inline in that type def:

`export type SmartDocumentsTemplateGroupWithParentId = Omit<
  GeneratedType<"RestSmartDocumentsTemplateGroup">,
  "groups" | "templates"
> & {
  groups?: SmartDocumentsTemplateGroupWithParentId[] | null;
  templates?:
    | (GeneratedType<"RestSmartDocumentsTemplate"> & {
        parentGroupId?: string | null;
      })
    | null;
};`

What is your pref?

Reply:
what does it do with the `Generated` type? Or is it copying the type definition and leaving information out? I don't like either way then.... I then prefer that we define it normally with the fields that we know we need in this type !

Reply:
I want @xiduzo and @edgarvonk to tell me if we should have this coupling between backend API type and frontend internal type. What is are their opinions?